src/NuGet.Clients/NuGet.PackageManagement.UI/Options/ConfigurationFilesOptionsPage.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Windows;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Configuration;
+using NuGet.ProjectManagement;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.UI.Options
+{
+    [Guid("C17B308A-00BB-446E-9212-2D14E1005985")]
+    public class ConfigurationFilesOptionsPage : UIElementDialogPage
+    {
+        private Lazy<ConfigurationFilesControl> _configurationFilesControl;
+
+        /// <summary>
+        /// Gets the Windows Presentation Foundation (WPF) child element to be hosted inside the Options dialog page.
+        /// </summary>
+        /// <returns>The WPF child element.</returns>
+        protected override UIElement Child => _configurationFilesControl.Value;
+
+        public ConfigurationFilesOptionsPage()
+        {
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            ISettings settings = componentModel.GetService<Configuration.ISettings>();
+            INuGetProjectContext projectContext = componentModel.GetService<INuGetProjectContext>();
+            _configurationFilesControl = new Lazy<ConfigurationFilesControl>(() => new ConfigurationFilesControl(new ConfigurationFilesViewModel(settings, projectContext)));
+        }
+
+        /// <summary>
+        /// This method is called when VS wants to activate this page.
+        /// ie. when the user opens the tools options page.
+        /// </summary>
+        /// <param name="e">Cancellation event arguments.</param>
+        protected override void OnActivate(CancelEventArgs e)
+        {
+            // The UI caches the settings even though the tools options page is closed.
```

Comment:
`OnActivate` gets called every time you select `Configuration Files` window, so this method will always return the latest data that was saved. This dialog must be closed in case you want to modify the file in VS.

If you modify the file outside VS you must refresh this window

Reply:
I think if you load the options window, then modify the config externally, the options window will overwrite the file. Probably worth testing, but also seems low frequency/risk.